### DISCUSSION TITLE
Is there a library/module for "Creating" PDFs in Nuxt/Vue?

### DISCUSSION THREAD

**Discussion by HenrijsS** [nuxt/nuxt#22045] [Questions]
The one thing that put me on making my project in next.js is [react-pdf](https://github.com/diegomura/react-pdf).

Is there any equivalent in the Vue ecosystem? Because Vercel is scummy and bloated my bundle-sizes so much, that CloudFlare pages refused to build it due to size limitations.

I have never had issues like these with Nuxt, but I really need a PDF **Creation** library.

**Comment by Amar-Gill** [comment]
For client side you can try:
- https://parallax.github.io/jsPDF/docs/jsPDF.html
- https://pdfkit.org/

My preferred method is server side using [puppeteer](https://pptr.dev/)
- https://pptr.dev/guides/pdf-generation

NuxtHub also has an example that uses CF browser rendering:
- https://hub.nuxt.com/docs/features/browser#pdf-generation

**Comment by xjccc** [comment]
Have a try
https://github.com/unjs/unpdf
